### PR TITLE
A bazel build and regression harness for s2geometry

### PR DESCRIPTION
**Pull request by spendres**
See issue #53. 

This version relies on the bazel central repository for abseil-cpp, googletest, and boringssl and points to HEAD of s2geometry.

### CODE REVIEW

File: src/s2/base/logging.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
+// Copyright Google Inc. All Rights Reserved.
```
Comment (jmr) [inline src/s2/base/logging.h:1]: This file should be deleted.

File: src/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,1962 @@
+# @s2geometry/BUILD.bazel
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
+package(default_visibility = ["//visibility:public"])
+
+cc_library(name = "s2-includes",includes = ["s2"],)
```
Comment (jmr) [inline src/BUILD.bazel:5]: what's this for? fix the formatting.

File: src/.bazelversion [hunk 3]
```diff
@@ -0,0 +1 @@
+7.1.1
```
Comment (jmr) [inline src/.bazelversion:1]: This looks like the latest release. What do we actually need?

File: src/README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+
+# S2 Geometry Library
```
Comment (jmr) [inline src/README.md:2]: This seems like a shortened version of https://github.com/google/s2geometry/blob/master/README.md
better to combine with that

File: src/BUILD.bazel [hunk 5]
```diff
@@ -0,0 +1,1962 @@
...
+
+cc_library(
+    name = "base",
```
Comment (jmr) [inline src/BUILD.bazel:8]: Can you explain the organization?  Why is this library here instead of in `src/s2/base/BUILD.bazel`?

**Review by jmr — CHANGES_REQUESTED** [review]

File: src/README.md (hunk 4 above)
Comment (spendres) [reply src/README.md]: 7.0.0 for bzlmod

File: src/BUILD.bazel (hunk 5 above)
Comment (spendres) [reply src/BUILD.bazel]: agreed

**Comment by spendres** [comment]
...logging.h was removed in most recent commit/push.

**Comment by spendres** [comment]
I tested build times for a hierarchical library definition as compared to a single flat library definition for s2. The flat definition builds twice as fast (12 sec vs 25 sec) about 25% of the time – using a mac M1. This is counter to expectations, so I reverted to the flat definition.

**Comment by spendres** [comment]
Thanks for the requested changes by mmorel-35 … made and pushed.

**Comment by jmr** [comment]
Thank you!

**Pull request merged**